std/experimental/typecons.d

```diff
@@ -1024,3 +1024,10 @@ pure nothrow @safe unittest
     static assert(!__traits(compiles, arr ~= 4));
     assert((arr ~ 4) == [1, 2, 3, 4]);
 }
+
+pure nothrow @nogc @system unittest
+{
+    int i = 1;
+    Final!(int*) fp = &i;
+    assert(*fp == 1);
```

Comment:
The concern is that it should not be some other wrapper type masquerading as int. I'd also check that `i` has the same value as `*fp` after the assignment.